temporal/clientoptions.go

```diff
@@ -51,6 +59,13 @@ func ClientOptions(cfg ClientConfig, log *zap.Logger) (client.Options, error) {
 		Logger:             logx.NewZapAdapter(log),
 		ContextPropagators: []workflow.ContextPropagator{NewContextPropagator()},
 	}
+	if cfg.UseTemporalCloud {
```

Comment:
nit: do we need 2 vars or could we do the following?
```suggestion
	if cfg.APIKey != "" {
```